packages/core/src/zone/ng_zone.ts

```diff
@@ -159,13 +162,17 @@ export class NgZone {
 
   static assertInAngularZone(): void {
     if (!NgZone.isInAngularZone()) {
-      throw new Error('Expected to be in Angular Zone, but it is not!');
+      throw new RuntimeError(
+          RuntimeErrorCode.INVALID_ZONE,
```

Comment:
I'd propose updating the error code a bit:
```suggestion
          RuntimeErrorCode.UNEXPECTED_ZONE_STATE,
```